tests/Integration/Core/LoadDiscoveryClassesTest.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Tempest\Integration\Core;
+
+use function PHPUnit\Framework\assertNotContains;
+use Tempest\Core\Kernel\LoadDiscoveryClasses;
+use Tempest\Database\DatabaseConfig;
+use Tempest\Database\MigrationDiscovery;
+use function Tempest\get;
+use Tests\Tempest\Fixtures\Discovery\HiddenMigration;
+use Tests\Tempest\Integration\FrameworkIntegrationTestCase;
+
+/**
+ * @internal
+ */
+final class LoadDiscoveryClassesTest extends FrameworkIntegrationTestCase
+{
+    public function test_hidden_from_discovery(): void
+    {
+        $this->kernel->discoveryClasses = [
+            MigrationDiscovery::class,
+        ];
+
+        $this->kernel->discoveryLocations = [
+            realpath(__DIR__.'../../Fixtures/Discovery'),
+        ];
+
+        (new LoadDiscoveryClasses($this->kernel, $this->container));
+
+        $migrations = get(DatabaseConfig::class)->getMigrations();
+
+        assertNotContains(HiddenMigration::class, $migrations);
```

Comment:
Just a style preference, but I'd use $this->assertNotContains, but it's not a blocker